networks/rpc/http_newrelic.go

```diff
@@ -0,0 +1,136 @@
+package rpc
+
+import (
+	"bytes"
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+	"runtime/debug"
+
+	"github.com/gin-gonic/gin/binding"
+	"github.com/newrelic/go-agent/v3/newrelic"
+)
+
+// dupWriter writes data to the buffer as well as http response
+type dupWriter struct {
+	http.ResponseWriter
+	body *bytes.Buffer
+}
+
+func (w dupWriter) Write(b []byte) (int, error) {
+	w.body.Write(b)
+	return w.ResponseWriter.Write(b)
+}
+
+// KASAttrs contains identifications for a KAS request
+type KASAttrs struct {
+	ChainID      string `header:"x-chain-id" json:"x-chain-id"`
+	AccountID    string `header:"x-account-id" json:"x-account-id"`
+	RequestID    string `header:"x-request-id" json:"x-request-id"`
+	ParentSpanID string `header:"x-b3-parentspanid,omitempty" json:"x-b3-parentspanid,omitempty"`
+	SpanID       string `header:"x-b3-spanid,omitempty" json:"x-b3-spanid,omitempty"`
+	TraceID      string `header:"x-b3-traceid,omitempty" json:"x-b3-traceid,omitempty"`
+}
+
+func parseKASHeader(r *http.Request) KASAttrs {
+	attrs := KASAttrs{}
+	if err := binding.Header.Bind(r, &attrs); err != nil {
+		logger.Error("failed to bind a KAS header")
+	}
+	return attrs
+}
+
+// newNewRelicHTTPHandler enables NewRelic web transaction monitor.
+// It also prints error logs when RPC returns contains error messages.
+func newNewRelicHTTPHandler(nrApp *newrelic.Application, handler http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer func() {
+			if err := recover(); err != nil {
+				logger.Error("NewRelic http handler panic\n"+string(debug.Stack()), "err", err)
+			}
+		}()
+
+		// parse RPC requests
+		reqs, isBatch, err := getRPCRequests(r)
+		if err != nil || len(reqs) < 1 {
+			logger.Error("failed to parse RPC request", "err", err, "len(reqs)", len(reqs))
```

Comment:
Do we need to leave logs for all invalid RPC requests?

Reply:
@jeongkyun-oh KAS team wants to print all RPC logs including success cases as well as RPC error logs. Let's consider the proper log level for RPC requests later.

Reply:
@aidan-kwon This PR prints logs for all success cases?
Which routine does print the success cases?

Reply:
@KimKyungup It doesn't print success cases. I mean we need to consider it later since KAS Team requested it.

Reply:
I think printing all logs is not a good idea also.